internal/spokes/spokes.go

```diff
@@ -88,18 +89,33 @@ func Exec(ctx context.Context, stdin io.Reader, stdout io.Writer, stderr io.Writ
 		config:           config,
 		statelessRPC:     *statelessRPC,
 		advertiseRefs:    *httpBackendInfoRefs,
-		quarantineFolder: filepath.Join(repoPath, "objects", quarantineID),
+		quarantineFolder: quarantineFolder,
 		governor:         g,
 	}
 
+	registerShutdownHooks(rp)
```

Comment:
Would it make sense to use a `context.Context` object to handle transmitting SIGINT or SIGTERM down into the spokes package? The spokes package is already wired to handle contexts, so it seems like the following would work, and would avoid the awkwardness of multiple exit points.

```suggestion
	ctx, cancel := signal.NotifyContext(ctx, syscall.SIGINT, syscall.SIGTERM)
	defer cancel()
```

Reply:
I think we had this covered here https://github.com/github/spokes-receive-pack/blob/044037d7a40636ed1be8efe568c0bc0534c83316/internal/spokes/spokes.go#L40 but I think we still need to deal with the signals at the `spokes-receive-pack` level, not "only" at the "ExecContext" level

I have tried to use the context instead of the original approach here 0016e37fa9afa8e64d5c324da9d774347b58164d.

This proposal doesn't change the fact that we are calling the `RemoveQuarantine` method at two different places. I could change the execute method to return a `chan error` so we can "translate" the `isDone` into an error and have a single place where we clean the quarantine, but, to be honest, I am not sure if it's worth it?

Reply:
Ah, good catch, I missed that signal handlers were set up in the spokes package already.